### PR TITLE
feat(LocalLambdaInvokeService): Command for start-lambda

### PR DESCRIPTION
**Pull request by jfuss**
Command help text:

```
Usage: samdev local start-lambda [OPTIONS]

  Allows you to run a Local Lambda Service that will service the invoke path
  to your functions for quick development &   testing through the AWS CLI or
  SDKs. When run in a directory that contains your Serverless functions and
  your AWS   SAM template, it will create a local HTTP server that wil
  response to the invoke call to your functions.   When accessed (via
  browser, cli etc), it will launch a Docker container locally to invoke the
  function. It will read   the CodeUri property of AWS::Serverless::Function
  resource to find the path in your file system containing the Lambda
  Function code. This could be the project's root directory for interpreted
  languages like Node & Python, or a build   directory that stores your
  compiled artifacts or a JAR file. If you are using a interpreted language,
  local changes   will be available immediately in Docker container on every
  invoke. For more compiled languages or projects requiring   complex
  packing support, we recommended you run your own building solution and
  point SAM to the directory or file   containing build artifacts.

Options:
  --host TEXT                     Local hostname or IP address to bind to
                                  (default: '127.0.0.1')
  -p, --port INTEGER              Local port number to listen on (default:
                                  '3000')
  -t, --template PATH             AWS SAM template file  [default:
                                  template.[yaml|yml]]
  -n, --env-vars PATH             JSON file containing values for Lambda
                                  function's environment variables.
  -d, --debug-port TEXT           When specified, Lambda function container
                                  will start in debug mode and will expose
                                  this port on localhost.
  --debug-args TEXT               Additional arguments to be passed to the
                                  debugger
  -v, --docker-volume-basedir TEXT
                                  Specifies the location basedir where the SAM
                                  file exists. If the Docker is running on a
                                  remote machine, you must mount the path
                                  where the SAM file exists on the docker
                                  machine and modify this value to match the
                                  remote machine.
  --docker-network TEXT           Specifies the name or id of an existing
                                  docker network to lambda dockercontainers
                                  should connect to, along with the default
                                  bridge network. If not specified,the Lambda
                                  containers will only connect to the default
                                  bridge docker network.
  -l, --log-file TEXT             logfile to send runtime logs to
  --skip-pull-image               Specify whether CLI should skip pulling down
                                  the latest Docker image for Lambda runtime
  --profile TEXT                  Specify which AWS credentials profile to
                                  use.
  --debug                         Turn on debug logging
  --help                          Show this message and exit.
```

*Description of changes:*
Added the command line interface for 'start-lambda'

What is left:
- [x] Unit test
- [ ] Integ tests
- [ ] Update 'help' message

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samcli/commands/local/cli_common/options.py [hunk 1]
```diff
@@ -55,6 +55,36 @@ def template_click_option():
...
+    -------
+    function
+        The callback function
+    """
+    service_options = [
+        click.option('--host',
+                     default="127.0.0.1",
+                     help="Local hostname or IP address to bind to (default: '127.0.0.1')"),
+        click.option("--port", "-p",
+                     default=3000,
```
Comment (sanathkr) [inline samcli/commands/local/cli_common/options.py]: `start-lambda` should start in a different port by default. So you can use it with `start-api` without extra configuration. What do you think?

File: samcli/commands/local/lib/local_lambda_service.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        # to the console or a log file. stderr from Docker container contains runtime logs and output of print
+        # statements from the Lambda function
+        service = LocalLambdaInvokeService(lambda_runner=self.lambda_runner,
+                                           port=self.port,
+                                           host=self.host,
+                                           stderr=self.stderr_stream)
+
+        service.create()
+
+        LOG.info("Starting the Locally Lambda Service. You can now invoke you Lambda Functions through the endpoint.")
```
Comment (sanathkr) [inline samcli/commands/local/lib/local_lambda_service.py]: `Locally` - typo. This message is also unclear when you say "you can invoke Lambda functions through the endpoint?" it is not clear

File: samcli/commands/local/start_lambda/cli.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
+"""
+CLI command for "local start-api" command
```
Comment (sanathkr) [inline samcli/commands/local/start_lambda/cli.py]: it is not start-api

File: samcli/commands/local/start_lambda/cli.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+from samcli.cli.main import pass_context, common_options as cli_framework_options
+from samcli.commands.local.cli_common.options import invoke_common_options, service_common_options
+from samcli.commands.local.cli_common.invoke_context import InvokeContext
+from samcli.commands.local.cli_common.user_exceptions import UserException
+from samcli.commands.local.lib.local_lambda_service import LocalLambdaService
+from samcli.commands.validate.lib.exceptions import InvalidSamDocumentException
+
+LOG = logging.getLogger(__name__)
+
+HELP_TEXT = """
```
Comment (sanathkr) [inline samcli/commands/local/start_lambda/cli.py]: This help text needs more work. It doesn't well explain how this command works.

File: samcli/commands/local/start_lambda/cli.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+"""
+
+
+@click.command("start-lambda", help=HELP_TEXT, short_help="Runs a Local Lambda Service (for the Invoke path only)")
+@service_common_options
+@invoke_common_options
+@cli_framework_options
+@pass_context
+def cli(ctx,
+        # start-api Specific Options
```
Comment (sanathkr) [inline samcli/commands/local/start_lambda/cli.py]: not start-api

File: samcli/commands/local/cli_common/options.py (hunk 1 above)
Comment (jfuss) [reply samcli/commands/local/cli_common/options.py]: Good call. I think just doing a +1 is enough. Not sure if the delta is a big deal. Let me know if you think something greater is better.

File: samcli/commands/local/lib/local_lambda_service.py (hunk 2 above)
Comment (jfuss) [reply samcli/commands/local/lib/local_lambda_service.py]: Updated but still don't think it describes what I want very well. Will think on this message more.

File: samcli/commands/local/start_lambda/cli.py (hunk 3 above)
Comment (jfuss) [reply samcli/commands/local/start_lambda/cli.py]: gah... I didn't copy start-api at all... ;)

File: samcli/commands/local/start_lambda/cli.py (hunk 5 above)
Comment (jfuss) [reply samcli/commands/local/start_lambda/cli.py]: :(

**Comment by jfuss** [comment]
@sanathkr I am ok with that.

**Pull request merged**